swatch-core/src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java

```diff
@@ -41,12 +38,13 @@
   // method
   @SuppressWarnings({"linelength", "java:S107"})
   Page<TallySnapshot>
-      findByAccountNumberAndProductIdAndGranularityAndServiceLevelAndUsageAndSnapshotDateBetweenOrderBySnapshotDate(
+      findByAccountNumberAndProductIdAndGranularityAndServiceLevelAndUsageAndBillingProviderAndSnapshotDateBetweenOrderBySnapshotDate(
```

Comment:
@kahowell thoughts on maybe using Specifications or QueryDSL after a certain threshold of parameters because my brain does not like reading method names this long

Reply:
Same, I discussed this with @kahowell as i noticed that this will get longer with billingProductId. I'm going to tackle this in my next PR for ENT-4843